subsys/net/buf.c

```diff
@@ -455,6 +455,9 @@ void net_buf_unref(struct net_buf *buf)
 	while (buf) {
 		struct net_buf *frags = buf->frags;
 		struct net_buf_pool *pool;
+		k_spinlock_key_t key;
+
+		pool = net_buf_pool_get(buf->pool_id);
 
```

Comment:
Shouldn't we place the spinlock here so that the double free check gets protected too?

Reply:
I think the reading operation can be completed by one instruction.

Reply:
Done.